Ignore cache when rewriting expunged file

<b>Not a candidate for upstream without additional
   debate</b>

The cache sometimes gets rewritten without considering
cyrus.expunge.  This patch just skips even considering
the records within cyrus.expunge when rewriting the cache.

It means a reconstruct is needed after an unexpunge, but
that's rare so I don't care.  Better than potential
corruption due to misplaced pointers.

I don't think this is the right long-term solution, but
it won't hurt anything either, so I don't really care if
it goes upstream.
Index: cyrus-imapd-2.3.10/imap/mailbox.c
===================================================================
--- cyrus-imapd-2.3.10.orig/imap/mailbox.c	2007-11-18 06:37:45.000000000 -0500
+++ cyrus-imapd-2.3.10/imap/mailbox.c	2007-11-18 06:38:38.000000000 -0500
@@ -2371,7 +2371,7 @@
 	r = process_records(mailbox, newexpungeindex, expunge_index_base,
 			    expunge_exists, deleted, &numdeleted,
 			    &quotadeleted, &numansweredflag, &numdeletedflag,
-			    &numflaggedflag, newcache, &new_cache_total_size,
+			    &numflaggedflag, NULL, NULL,
 			    -1, 0, decideproc, deciderock, EXPUNGE_CLEANUP);
 	if (r) goto fail;
 	expunge_exists -= (numdeleted - new_deleted);
